patches_the_bot.py

```diff
@@ -1,132 +1,100 @@
 
-pubg = reddit.subreddit('pubattlegrounds')
-heroes = reddit.subreddit('heroesofthestorm')
-lol = reddit.subreddit('leagueoflegends')
-wow = reddit.subreddit('wow')
+    savedHrefs = patches.getSavedHrefs() 
 
-def main():
-    getFoundPosts() 
     logging.info("Initiating search")
 
     # PUBG
     logging.info("Finding patch notes for PUBG...")
-    submissions = pubg.submissions(start=submission_interval)
-    for submission in submissions:
-        if ('Early Access -'.lower() in submission.title.lower() 
-            and 'Update'.lower() in submission.title.lower() 
-            and 'http://steamcommunity.com/games/578080/announcements' in submission.url.lower()):
-            if submission.id not in foundPosts: 
-                submit("PlayerUnknown's Battlegrounds", "PC", submission)
+    soup = patches.makeSoup(gd.pubg_url) 
+    posts = soup.find_all("a", class_="large_title")
+    for post in posts: 
+        href = post['href']
+        name = post.string
+        if(all(string in name.lower() for string in gd.pubg_conditions)):
+            if(href not in savedHrefs):
+                patches.submit("PlayerUnknown's Battlegrounds", "PC", href, name)
             else: 
-                logging.info("Most recent post for PUBG already found...")
-            break    
+                logging.info(name + " already found!")
     # HEROES
     logging.info("Finding patch notes for HotS...")
-    submissions = heroes.submissions(start=submission_interval)
-    for submission in submissions:
-        if ('us.battle.net'.lower() in submission.url.lower() 
-            and 'heroes-of-the-storm-patch-notes' in submission.url.lower()):
-            if submission.id not in foundPosts: 
-                submit("Heroes of the Storm", "PC", submission)
+    soup = patches.makeSoup(gd.hots_url) 
+    posts = soup.find_all(class_="news-list__item__title")
+    for post in posts:
+        href = "http://us.battle.net" + post.find('a')['href']
+        name = post.find('a').string.replace('\n', '')
+        if(any(string in href.lower() for string in gd.hots_conditions)):
+            if(href not in savedHrefs): 
+                logging.info("Submitting: " + name)
+                patches.submit("Heroes of the Storm", "PC", href, name) 
             else:
-                logging.info("Most recent post for HotS already found...")
-            break
+                logging.info(name + " already found!")
     # LoL
     logging.info("Finding patch notes for LoL...")
-    submissions = lol.submissions(start=submission_interval)
-    for submission in submissions:
-        if ('leagueoflegends.com/en/news/game-updates/patch/'.lower() in submission.url.lower()):
-            if submission.id not in foundPosts: 
-                submit("League of Legends", "PC", submission)
+    soup = patches.makeSoup(gd.lol_url) 
+    posts = soup.find_all("h4")
+    for post in posts:
+        href = "http://eune.leagueoflegends.com/" + post.find('a')['href']
```

Comment:
Remove following '/' from stringUrl so that it doesn't get saved as 'stringUrl//url' in the future (not a breaking issue)